_pylibczi/_pylibczi.cpp

```diff
@@ -207,6 +228,147 @@ static PyObject *cziread_allsubblocks(PyObject *self, PyObject *args) {
+    rect->y = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 1)));
+    rect->w = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 2)));
+    rect->h = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 3)));
+    if( PyErr_Occurred() != NULL) { // collectively check if an PyErr occurred
+        PyErr_SetString(PylibcziError, "problem converting region=(x: [Int], y: [Int], w: [Int], h: [Int])");
+        return 0;
+    }
+    return 1;
+}
+
+bool isValidRegion(const libCZI::IntRect &inBox, const libCZI::IntRect &cziBox ){
+    bool ans = true;
+    // check origin is in domain
+    if( inBox.x < cziBox.x || cziBox.x + cziBox.w < inBox.x) ans = false;
+    if( inBox.y < cziBox.y || cziBox.y + cziBox.h < inBox.y) ans = false;
+
+    // check  (x1, y1) point is in domain
+    int x1 = inBox.x + inBox.w;
+    int y1 = inBox.y + inBox.h;
+    if( x1 < cziBox.x || cziBox.x + cziBox.w < x1) ans = false;
+    if( y1 < cziBox.y || cziBox.y + cziBox.h < y1) ans = false;
+
+    if(!ans) {
+        std::cerr << "Conflict region requested("
+                  << inBox.x << ", " << inBox.y << ", " << inBox.w << ", " << inBox.h
+                  << ") in plane ("
+                  << cziBox.x << ", " << cziBox.y << ", " << cziBox.w << ", " << cziBox.h << ")!"
+                  << std::endl;
+    }
+
+    if(inBox.w < 1 || 1 > inBox.h){
+        std::cerr << "Ill-defined region, with and heigh must be positive values! ("
+                << inBox.x << ", " << inBox.y << ", " << inBox.w << ", " << inBox.h << ")" << std::endl;
+        ans = false;
+    }
+
+    return ans;
+}
+
+static PyObject *cziread_mosaic(PyObject *self, PyObject *args) {
+    char *filename_buf;
+    float scaleFactor;
+    libCZI::CDimCoordinate planeCoord;
+    libCZI::IntRect imBox{0,0,0,0};
+    // parse arguments
+    if (!PyArg_ParseTuple(args, "sO&O&f", &filename_buf,
+                          &convertDictToPlaneCoords, &planeCoord,
+                          &listToIntRect, &imBox,
+                          &scaleFactor)
+            ){
+        PyErr_SetString(PylibcziError, "Error: conversion of arguments failed. Check arguments." );
+        return nullptr;
+    }
+
+    auto cziReader = open_czireader_from_cfilename(filename_buf);
+    auto statistics = cziReader->GetStatistics();
+
+    // handle the case where the function was called with region=None (default to all)
+    if ( imBox.w == -1 && imBox.h == -1 ) imBox = statistics.boundingBox;
+    if (!isValidRegion(imBox, statistics.boundingBox)){
+        PyErr_SetString(PylibcziError, "Error:  region not contained in specified CZI plane." );
+        return nullptr;
+    }
+
+
+    std::map< libCZI::DimensionIndex, std::pair< int, int> > limitTbl;
+
+    statistics.dimBounds.EnumValidDimensions([&limitTbl](libCZI::DimensionIndex di, int start, int size)->bool{
+        limitTbl.emplace( di, std::make_pair(start, size));
+        return true;
```

Comment:
So either `return true` or `return img`? I might be understanding this wrong but this seems inconsistent.

Reply:
that is the return value of the lambda passed in on line 351, not the return value of this function :)

Reply:
Oh oh I totally missed the lambda, thanks :)